packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts

```diff
@@ -131,17 +132,19 @@ function isBareModuleId(id: string): boolean {
 
 function warnOnceIfEsmOnlyPackage(
   packageName: string,
-  path: string,
+  fullImportPath: string,
   onWarning: (msg: string, key: string) => void
 ) {
-  let packageJsonFile = require.resolve(`${packageName}/package.json`);
-  if (!fs.existsSync(packageJsonFile)) {
-    console.log(packageJsonFile, `does not exist`);
+  let packageDir = resolveModuleBasePath(packageName);
+  let pkgJsonPath = path.join(packageDir, "package.json");
+
+  if (!fs.existsSync(pkgJsonPath)) {
+    console.log(pkgJsonPath, `does not exist`);
     return;
   }
-  let pkg = JSON.parse(fs.readFileSync(packageJsonFile, "utf8"));
+  let pkg = JSON.parse(fs.readFileSync(pkgJsonPath, "utf-8"));
```

Comment:
I'm not a maintainer so you can ignore this comment, but I don't think abbreviations add value anywhere.